resources/patches/scheduler_workbench-allowed_status.patch

```diff
@@ -0,0 +1,147 @@
+diff --git a/scheduler_workbench.admin.inc b/scheduler_workbench.admin.inc
+new file mode 100755
+index 0000000..403f006
+--- /dev/null
++++ b/scheduler_workbench.admin.inc
+@@ -0,0 +1,18 @@
++<?php
++
++/**
++ * Form callback for the settings form.
++ *
++ * @see system_settings_form()
++ */
++function scheduler_workbench_allowed_status_form($form, &$form_state) {
++  $options = workbench_moderation_state_labels();
++  $form['scheduler_workbench_allowed_status'] = array(
++    '#type' => 'checkboxes',
++    '#title' => t('Allowed Workbench Moderation status for scheduling publication'),
++    '#options' => $options,
++    '#default_value' => variable_get('scheduler_workbench_allowed_status', array()),
++    '#description' => t('Choose status to allow revisions for scheduling publication. When Scheduler Workbench tries to publish a revision i will check if the current state is one of the selected from this list.'),
++  );
++  return system_settings_form($form);
++}
+diff --git a/scheduler_workbench.install b/scheduler_workbench.install
+new file mode 100755
+index 0000000..1bfe0c4
+--- /dev/null
++++ b/scheduler_workbench.install
+@@ -0,0 +1,46 @@
++<?php
++
++/**
++ * @file
++ * Install, update and uninstall functions for the scheduler_workbench module.
++ */
++
++/**
++ * Implements hook_install().
++ */
++function scheduler_workbench_install() {
++  _scheduler_workbench_status();
++}
++
++/**
++ * Implements hook_uninstall().
++ */
++function scheduler_workbench_uninstall() {
++  variable_del('scheduler_workbench_allowed_status');
++}
++
++/**
++ * Implements hook_update_N().
++ */
++function scheduler_workbench_update_7001() {
++  if (!variable_set('scheduler_workbench_allowed_status', array())) {
++    _scheduler_workbench_status();
++  }
++}
++
++/**
++ * Sets the allowed status that can be scheduled for publication.
++ */
++function _scheduler_workbench_status() {
++  $transitions = workbench_moderation_transitions();
++  $status = array();
++  foreach ($transitions as $key => $value) {
++    if ($value->to_name != 'published') {
++      unset($transitions[$key]);
```

Comment:
This interaction seems unnecessary, can't we simply add $value->from_name to the $status array if the condition is met?
`if ($value->to_name == 'published') {
  $status[] = $value->from_name;
}
`